docs/glossary.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: default
+title: lakeFS Glossary
+description: Glossary of all terms related to lakeFS technical internals and the architecture
+nav_order: 52
+has_children: false
+---
+
+# Glossary
+
+{% include toc.html %}
+
+### Auditing
```

Comment:
Should we explain how auditing is related to lakeFS? as a reader, I'm not sure that I understand why the lakeFS glossary tells me about auditing. WDYT?

Reply:
Makes sense. I will add - lakeFS commit history helps data teams with auditing the data.